pkg/controller/extensions/extension/extension_test.go

```diff
@@ -0,0 +1,600 @@
+						Type:    corev1alpha1.Failed,
+						Status:  corev1.ConditionTrue,
+						Reason:  reasonCreatingRBAC,
+						Message: fmt.Errorf("failed to create service account: %+v", errors.New("test-create-sa-error")).Error(),
+					})),
+			},
+		},
+		{
+			name: "FailDeployment",
+			r: resource(
+				withPolicyRules(defaultPolicyRules()),
+				withControllerSpec(defaultControllerSpec())),
+			clientFunc: func(r *v1alpha1.Extension) client.Client {
+				return &test.MockClient{
+					MockCreate: func(ctx context.Context, obj runtime.Object) error {
+						if _, ok := obj.(*apps.Deployment); ok {
+							return errors.New("test-create-deployment-error")
+						}
+						return nil
+					},
+					MockStatusUpdate: func(ctx context.Context, obj runtime.Object) error { return nil },
+				}
+			},
+			want: want{
+				result: resultRequeue,
+				err:    nil,
+				r: resource(
+					withPolicyRules(defaultPolicyRules()),
+					withControllerSpec(defaultControllerSpec()),
+					withConditions(corev1alpha1.Condition{
+						Type:    corev1alpha1.Failed,
+						Status:  corev1.ConditionTrue,
+						Reason:  reasonCreatingDeployment,
+						Message: fmt.Errorf("failed to create deployment: %+v", errors.New("test-create-deployment-error")).Error(),
+					})),
+			},
+		},
+		{
+			name:       "SuccessfulCreate",
+			r:          resource(),
+			clientFunc: func(r *v1alpha1.Extension) client.Client { return fake.NewFakeClient(r) },
+			want: want{
+				result: requeueOnSuccess,
+				err:    nil,
+				r: resource(
+					withConditions(corev1alpha1.Condition{Type: corev1alpha1.Ready, Status: corev1.ConditionTrue})),
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			handler := &extensionHandler{
+				kube: tt.clientFunc(tt.r),
+				i:    tt.r,
+			}
+
+			got, err := handler.create(ctx)
+
+			if diff := cmp.Diff(err, tt.want.err, test.EquateErrors()); diff != "" {
+				t.Errorf("create() want error != got error:\n%s", diff)
+			}
+
+			if diff := cmp.Diff(got, tt.want.result); diff != "" {
+				t.Errorf("create() got != want:\n%v", diff)
+			}
+
+			if diff := cmp.Diff(tt.r, tt.want.r); diff != "" {
+				t.Errorf("create() got != want:\n%v", diff)
+			}
+		})
+	}
+}
+
+// ************************************************************************************************
+// TestProcessRBAC
+// ************************************************************************************************
+func TestProcessRBAC(t *testing.T) {
+	type want struct {
+		err error
+		sa  *corev1.ServiceAccount
+		cr  *rbac.ClusterRole
+		crb *rbac.ClusterRoleBinding
+	}
+
+	tests := []struct {
+		name       string
+		r          *v1alpha1.Extension
+		clientFunc func(*v1alpha1.Extension) client.Client
+		want       want
+	}{
+		{
+			name:       "NoPermissionsRequested",
+			r:          resource(),
+			clientFunc: func(r *v1alpha1.Extension) client.Client { return fake.NewFakeClient(r) },
+			want: want{
+				err: nil,
```

Comment:
Nit: Here and everywhere, by emitting `field: nil`, the field value will be `nil` implicitly. 
Consider:
```suggestion
				watn: want{},
```